src/test/regress/expected/multi_repartition_udt.out

```diff
@@ -0,0 +1,219 @@
+ master_create_worker_shards 
+-----------------------------
+ 
+(1 row)
+
+INSERT INTO repartition_udt values (1, '(1,1)'::test_udt, 'foo');
+INSERT INTO repartition_udt values (2, '(1,2)'::test_udt, 'foo');
+INSERT INTO repartition_udt values (3, '(1,3)'::test_udt, 'foo');
+INSERT INTO repartition_udt values (4, '(2,1)'::test_udt, 'foo');
+INSERT INTO repartition_udt values (5, '(2,2)'::test_udt, 'foo');
+INSERT INTO repartition_udt values (6, '(2,3)'::test_udt, 'foo');
+INSERT INTO repartition_udt_other values (7, '(1,1)'::test_udt, 'foo');
+INSERT INTO repartition_udt_other values (8, '(1,2)'::test_udt, 'foo');
+INSERT INTO repartition_udt_other values (9, '(1,3)'::test_udt, 'foo');
+INSERT INTO repartition_udt_other values (10, '(2,1)'::test_udt, 'foo');
+INSERT INTO repartition_udt_other values (11, '(2,2)'::test_udt, 'foo');
+INSERT INTO repartition_udt_other values (12, '(2,3)'::test_udt, 'foo');
+SET client_min_messages = LOG;
+-- Query that should result in a repartition join on int column, and be empty.
+SELECT * FROM repartition_udt JOIN repartition_udt_other
+    ON repartition_udt.pk = repartition_udt_other.pk
+	WHERE repartition_udt.pk > 1;
+ pk | udtcol | txtcol | pk | udtcol | txtcol 
+----+--------+--------+----+--------+--------
+(0 rows)
+
+-- Query that should result in a repartition join on UDT column.
+SET citus.large_table_shard_count = 1;
+SET citus.task_executor_type = 'task-tracker';
+SET citus.log_multi_join_order = true;
+SET citus.explain_all_tasks = true;
+EXPLAIN SELECT * FROM repartition_udt JOIN repartition_udt_other
+    ON repartition_udt.udtcol = repartition_udt_other.udtcol
+	WHERE repartition_udt.pk > 1;
+LOG:  join order: [ "repartition_udt" ][ dual partition join "repartition_udt_other" ]
+                               QUERY PLAN                                
+-------------------------------------------------------------------------
+ Distributed Query into pg_merge_job_535003
+   Executor: Task-Tracker
+   Task Count: 4
+   Tasks Shown: None, not supported for re-partition queries
+   ->  MapMergeJob
+         Map Task Count: 3
+         Merge Task Count: 4
+   ->  MapMergeJob
+         Map Task Count: 5
+         Merge Task Count: 4
+ Master Query
+   ->  Seq Scan on pg_merge_job_535003  (cost=0.00..0.00 rows=0 width=0)
+(12 rows)
+
+SET citus.explain_all_tasks = false;
+SELECT * FROM repartition_udt JOIN repartition_udt_other
+    ON repartition_udt.udtcol = repartition_udt_other.udtcol
+	WHERE repartition_udt.pk > 1
+	ORDER BY repartition_udt.pk;
+LOG:  join order: [ "repartition_udt" ][ dual partition join "repartition_udt_other" ]
+ pk | udtcol | txtcol | pk | udtcol | txtcol 
+----+--------+--------+----+--------+--------
+  2 | (1,2)  | foo    |  8 | (1,2)  | foo
+  3 | (1,3)  | foo    |  9 | (1,3)  | foo
+  4 | (2,1)  | foo    | 10 | (2,1)  | foo
+  5 | (2,2)  | foo    | 11 | (2,2)  | foo
+  6 | (2,3)  | foo    | 12 | (2,3)  | foo
+(5 rows)
+
+SET citus.log_multi_join_order = false;
```

Comment:
We don't really need to reset things at the end of the file, since each file is executed in a different session.